pandas/tests/frame/test_sorting.py

```diff
@@ -227,6 +227,21 @@ def test_stable_descending_multicolumn_sort(self):
                                    kind='mergesort')
         assert_frame_equal(sorted_df, expected)
 
+    @pytest.mark.parametrize('seed', list(range(100)))
+    def test_sort_multi_index(self, seed):
+        # GH 25775
+        pd.np.random.seed(seed)
```

Comment:
Using a single deterministic example is probably easier to read and maintain.

I've found that the following example reproduces the issue on 0.23.0:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '0.23.0'

In [2]: df = pd.DataFrame({'a': [0.4, 0.1, 0.3, 0.2], 'b': [0, 1, 1, 1], 'c': range(4), 'd': list('abcd')})

In [3]: df.set_index(list('abc')).sort_index(level=list('ba'))
Out[3]:
         d
a   b c
0.4 0 0  a
0.1 1 1  b
0.3 1 2  c
0.2 1 3  d
```

And works on master:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '0.25.0.dev0+596.g20d0ad159a'

In [2]: df = pd.DataFrame({'a': [0.4, 0.1, 0.3, 0.2], 'b': [0, 1, 1, 1], 'c': range(4), 'd': list('abcd')})

In [3]: df.set_index(list('abc')).sort_index(level=list('ba'))
Out[3]:
         d
a   b c
0.4 0 0  a
0.1 1 1  b
0.2 1 3  d
0.3 1 2  c
```

Reply:
Ok! I updated to a single example. I'm using the one given in #25775.